app/views/questionnaire.py

```diff
@@ -436,7 +436,7 @@ def submit_answers(routing_path, eq_id, form_type, schema):
     _store_submitted_time_in_session(submitted_time)
 
     if is_view_submitted_response_enabled(schema.json):
-        _store_viewable_submission(list(answer_store), metadata, submitted_time)
+        _store_viewable_submission(answer_store.answer_map, metadata, submitted_time)
```

Comment:
This should probably be in the another pull request as it's not related to saving the new compressed questionnaire state.